src/Sulu/Bundle/MediaBundle/Controller/MediaStreamController.php

```diff
@@ -274,4 +293,20 @@ protected function getStorage()
 
         return $this->storage;
     }
+
+    private function createLock(string $key): ?Lock
+    {
+        if (!class_exists(Factory::class)) {
+            // for locking the symfony/lock is required if not installed the media stream is not locked
+            return null;
+        }
+
+        if (SemaphoreStore::isSupported()) {
+            $store = new SemaphoreStore();
+        } else {
+            $store = new FlockStore();
+        }
+
+        return (new Factory($store))->createLock($key);
```

Comment:
If I read about the lock component in [this blog](https://symfony.com/blog/new-in-symfony-3-3-lock-component) I see that there should also be a service. Why aren't we using the service but creating the factory class and so on ourselves?

Reply:
And this implementation would not allow to e.g. use the `RedisStore`, which would maybe be necessary if we have a distributed environment.

Reply:
See no possibility to use a `key` based locking over the service just a general lock?

Thats by design so as thumbnails by default are generated per server and should not block other servers.

Reply:
From what I see in executing the `debug:container` command is that there is a `lock.factory` service, which is a class of type `Symfony\Component\Lock\Factory`. There you should be able to pass a key, shouldn't you?

There seems to be no documentation about this yet, but there is some explanation in [this PR](https://github.com/symfony/symfony/pull/22113).

Reply:
Any reason not to use the `lock.factory` service?